serverless/dashboard
# Thread safety of Python base SDK

### Description
* Related issue https://linear.app/serverless/issue/SC-742/python-sdk-thread-safety
* I've reviewed the code base and identified potentially problematic access to the tags dictionary
* Refactored tests to make sure each test uses the `serverlessSdk` in isolation to ensure no side-effects between tests.

### Testing done
Created multithreaded + async tests with concurrency to ensure we can concurrently 
* create spans
* set/update tags
* capture errors/warnings
